cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,26 +38,47 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
 
-	if !cmdUtil.IsValidProfile(profileName) {
+	//need to check if the current active profile has a VM
+	if cmdUtil.DoesVMExist(config.AllInstancesConfig.ActiveProfile) {
+		err := unsetCurrentOcContext()
+		if err != nil {
+			fmt.Println(err.Error())
+		}
```

Comment:
I think we should remove the function unsetCurrentOcContext() and directly call` cmdUtil.RemoveCurrentContext()` here as below. As unsetCurrentOcContext() not adding any value here. Also printing the error for glog.V(2)  is the right thing to do as user do not need to see this and we are not failing the code execution here.   
```
err := cmdUtil.RemoveCurrentContext()
if err != nil {
	if glog.V(2) {
		fmt.Println(fmt.Sprintf("%s", err.Error()))
	}
```

Reply:
+1 to call RemoveCurrentContext directly.

I was making it a function because earlier the VMExists check was part of unsetCurrentContext, we are not doing anything extra in there.